src/dune/coq_rules.ml

```diff
@@ -371,3 +372,50 @@ let coqpp_rules ~sctx ~build_dir ~dir (s : Dune_file.Coqpp.t) =
     Command.run ~dir:(Path.build build_dir) cc.coqpp args
   in
   List.map ~f:mlg_rule s.modules
+
+let extract_rules ~sctx ~build_dir ~dir ~dir_contents
+    (s : Dune_file.Coq_extract.t) =
+  let cc = create_ccoq sctx ~dir in
+  let expander = SC.expander sctx ~dir in
+  let coq_flags = s.flags in
+  let scope = SC.find_scope_by_dir sctx dir in
+  let coq_lib_db = Scope.coq_libs scope in
+  (* Coq flags for depending libraries *)
+  let theories_deps =
+    Coq_lib.DB.requires_for_user_written coq_lib_db s.theories
+  in
+  let theories_flags =
+    Command.of_result_map theories_deps ~f:(fun libs ->
+        Command.Args.S (List.concat_map libs ~f:setup_theory_flag))
+  in
+  let coq_module =
+    let coq = Dir_contents.coq dir_contents in
+    Coq_sources.extract coq s
+  in
+  let wrapper_name = "dummy" in
```

Comment:
This is obviously wrong, but I'm not sure what wrapper we need here.

Reply:
A dummy wrapper would be fine, as we are using `-R` to compile the libraries; wrapper is only relevant for users of the library which is not relevant here.

Most users so far just add the `Extract` module to the library [so this way `coq.extract` would become a field of `coq.theory`], but IMO we don't need that extra flexibility.

Another choice is to run the extract file with `coqtop`, so you don't even produce a `.vo` file; but feels a bit ugly I guess.

Reply:
Note that when you call `coqtop` or just `coqc` , wrapper name is set to something such as `Top` , which is similar to dummy as it is never used [except if the user compiles a file for real and then tries to use it]

You maybe we could set as a wrapper name `DuneExtraction` here

Reply:
Good idea.